lib/compat/wordpress-6.5/class-wp-navigation-block-renderer.php

```diff
@@ -0,0 +1,611 @@
+				$has_submenus = true;
+			}
+		}
+		return $has_submenus;
+	}
+
+	/**
+	 * Determine whether to load the view script.
+	 *
+	 * @param array         $attributes   The block attributes.
+	 * @param WP_Block_List $inner_blocks The list of inner blocks.
+	 * @return bool Returns whether or not to load the view script.
+	 */
+	private static function should_load_view_script( $attributes, $inner_blocks ) {
+		$has_submenus       = static::has_submenus( $inner_blocks );
+		$is_responsive_menu = static::is_responsive( $attributes );
+		return ( $has_submenus && ( $attributes['openSubmenusOnClick'] || $attributes['showSubmenuIcon'] ) ) || $is_responsive_menu;
+	}
+
+	/**
+	 * Returns the markup for a single inner block.
+	 *
+	 * @param WP_Block $inner_block The inner block.
+	 * @return string Returns the markup for a single inner block.
+	 */
+	private static function get_markup_for_inner_block( $inner_block ) {
+		$inner_block_content = $inner_block->render();
+		if ( ! empty( $inner_block_content ) ) {
+			if ( in_array( $inner_block->name, static::$needs_list_item_wrapper, true ) ) {
+				return '<li class="wp-block-navigation-item">' . $inner_block_content . '</li>';
+			}
+
+			return $inner_block_content;
+		}
+	}
+
+	/**
+	 * Returns the html for the inner blocks of the navigation block.
+	 *
+	 * @param array         $attributes   The block attributes.
+	 * @param WP_Block_List $inner_blocks The list of inner blocks.
+	 * @return string Returns the html for the inner blocks of the navigation block.
+	 */
+	private static function get_inner_blocks_html( $attributes, $inner_blocks ) {
+		$has_submenus            = static::has_submenus( $inner_blocks );
+		$should_load_view_script = static::should_load_view_script( $attributes, $inner_blocks );
+
+		$style                = static::get_styles( $attributes );
+		$class                = static::get_classes( $attributes );
+		$container_attributes = get_block_wrapper_attributes(
+			array(
+				'class' => 'wp-block-navigation__container ' . $class,
+				'style' => $style,
+			)
+		);
+
+		$inner_blocks_html = '';
+		$is_list_open      = false;
+
+		foreach ( $inner_blocks as $inner_block ) {
+			$is_list_item = in_array( $inner_block->name, static::$list_item_nav_blocks, true );
+
+			if ( $is_list_item && ! $is_list_open ) {
+				$is_list_open       = true;
+				$inner_blocks_html .= sprintf(
+					'<ul %1$s>',
+					$container_attributes
+				);
+			}
+
+			if ( ! $is_list_item && $is_list_open ) {
+				$is_list_open       = false;
+				$inner_blocks_html .= '</ul>';
+			}
+
+			$inner_blocks_html .= static::get_markup_for_inner_block( $inner_block );
+		}
+
+		if ( $is_list_open ) {
+			$inner_blocks_html .= '</ul>';
+		}
+
+		// Add directives to the submenu if needed.
+		if ( $has_submenus && $should_load_view_script ) {
+			$tags              = new WP_HTML_Tag_Processor( $inner_blocks_html );
+			$inner_blocks_html = block_core_navigation_add_directives_to_submenu( $tags, $attributes );
```

Comment:
Why do we do this `$tags` work outside of the function?

Reply:
`block_core_navigation_add_directives_to_submenu` is a pre-existing function. If we move $tags inside it we'll need to change the signature. Is that safe to do or should we be concerned that someone might be using `block_core_navigation_add_directives_to_submenu`?

Reply:
Let's leave it. Thank you!